### ISSUE TITLE
gargoyle doesn't die sometimes when it inherits TCP sockets

### ISSUE THREAD

**Issue by danbornside** [obsidiansystems/gargoyle#10]
gargoyle doesn't die sometimes when it inherits TCP sockets

**Comment by mikereinhart** [comment]
I believe I also observed this. I killed https://gitlab.com/obsidian.systems/tezos-bake-monitor using Control+C, but gargoyle-nix-postres was left behind. Attempting to restart Kiln failed due to the left behind gargoyle process. Manually killing gargoyle allowed me to successfully start Kiln

**Comment by 3noch** [comment]
We also saw a bewildering issue where HTTP requests from Kiln timed out until we killed the lingering gargoyle+postgres.

**Comment by ryantrinkle** [comment]
@danharaj Any thoughts on this?

**Comment by danharaj** [comment]
No, I'd need to regain my bearings on this code in order to diagnose. Does anyone have a way to reliably reproduce this. Do @mikereinhart 's steps always work?

**Comment by mikereinhart** [comment]
In my experiences they do not always work. The issue is intermittent

**Comment by 3noch** [comment]
@danbornside told me that he has seen this issue when starting gargoyle postgres *after* binding a port (such that it gets inherited). I have not confirmed that this is a reliable way to reproduce.
